File: Free_ChatGPT/sistemaTipos/tipos.c
#include <stdio.h>
#include <stdlib.h>

double precioFinal(double precio, const char *descuento) {
    double porcentaje = atof(descuento);
    return precio * (1 - porcentaje / 100.0);
}

int main() {
    double resultado = precioFinal(50, "10");
    printf("%.2f\n", resultado);
    return 0;
}

File: Free_ChatGPT/sistemaTipos/tipos.rs
fn precio_final(precio: f64, descuento: &str) -> f64 {
    let descuento: f64 = descuento.parse().unwrap();
    precio * (1.0 - descuento / 100.0)
}

fn main() {
    let resultado = precio_final(50.0, "10");
    println!("{}", resultado);
}